packages/core/src/package/analyser/FHTAnalyzer.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import { ComponentSet, registry } from '@salesforce/source-deploy-retrieve';
+import SfpPackage, { PackageType } from '../SfpPackage';
+import { PackageAnalyzer } from './PackageAnalyzer';
+
+export default class FHTAnalyser implements PackageAnalyzer{
+    
+    public async analyze(sfpPackage: SfpPackage): Promise<SfpPackage> {
+        let isFHTFieldFound = false;
+        let fhtFields: { [key: string]: Array<string> } = {};
+
+        //read the yaml
+        let fhtYamlPath = path.join(
+            sfpPackage.workingDirectory,
+            sfpPackage.projectDirectory,
+            sfpPackage.packageDirectory,
+            '/postDeploy/history-tracking.yaml'
+        );
+
+        //read components mentioned in yaml
+        if (fs.existsSync(fhtYamlPath)) {
+            //convert yaml to json
+            fhtFields = yaml.load(fs.readFileSync(fhtYamlPath, { encoding: 'utf-8' }));
+        }
+
+        //Load component Set
+        let componentSet = ComponentSet.fromSource(
```

Comment:
if we pass the componentSet from the builder, we only need to generate it once and all other analysers/other possible functions in the builder can use it